src/db/spanner/support.rs

```diff
@@ -40,17 +40,13 @@ impl ToSpannerValue for String {
 
 impl ToSpannerValue for i32 {
     fn to_spanner_value(&self) -> Value {
-        let mut value = Value::new();
-        value.set_number_value(*self as f64);
-        value
+        self.to_string().to_spanner_value()
```

Comment:
IIRC, our `SpannerDb` implementation in sync uses type hints `TypeCode` for queries at least.
e.g. https://github.com/mozilla-services/syncstorage-rs/blob/master/src/db/spanner/models.rs#L977

We might want to verify that for query instances where we do this conversion have the type set.

Reply:
I think that's what this is doing? https://github.com/mozilla-services/syncstorage-rs/blob/c5da264a37196f85b116df9a66c681c0b65a2dc2/src/db/spanner/support.rs#L68

Reply:
Now I'm wondering if we should just iterate through the params list to build the types much like how we iterate through the params to convert to spanner values. It would set a fair pattern for handling parameters. Bonus would be some impl we could call, pass the parameters to, and it would automagically gen the `.parameters()` and `parameter_types()`...

Reply:
I like your idea about generating the type hints from the parameter name 🤔 At the very least, I'll be sure to add the type hints for the queries that involve the `ToSpannerValue` trait, but I'll see if I can figure something fancier out